File: frontend/src/types/medicalRecord.ts
export interface MedicalRecord {
  id: number;
  patient_id: number;
  doctor_id: number;
  appointment_id: number;
  diagnosis: string;
  symptoms: string;
  treatment: string;
  notes: string;
  created_at: string;
}

export interface MedicalRecordCreate {
  appointment_id: number;
  diagnosis: string;
  symptoms: string;
  treatment: string;
  notes: string;
}

export interface MedicalRecordUpdate {
  diagnosis?: string;
  symptoms?: string;
  treatment?: string;
  notes?: string;
}

File: app/schemas/medical_record.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MedicalRecordCreate(BaseModel):
    appointment_id: int
    diagnosis: str
    symptoms: str
    treatment: str
    notes: str


class MedicalRecordResponse(BaseModel):
    id: int
    patient_id: int
    doctor_id: int
    appointment_id: int
    diagnosis: str
    symptoms: str
    treatment: str
    notes: str
    created_at: datetime


class MedicalRecordUpdate(BaseModel):
    diagnosis: Optional[str] = None
    symptoms: Optional[str] = None
    treatment: Optional[str] = None
    notes: Optional[str] = None